dbt_platform_helper/commands/environment.py

```diff
@@ -572,50 +531,52 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
 
-    for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+    for index, svc in enumerate(services):
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        user_ip = get_public_ip()
+        allowed_ips = list(allowed_ips) + [user_ip]
+        for ip_index, ip in enumerate(allowed_ips):
+            forwarded_rule_priority = (index + 1) * (ip_index + 100)
```

Comment:
@acodeninja can you remember if this is still needed?

Reply:
It might not be, this was put in to ensure enough space between priorities as two entries cannot have the same priority values. From the looks of it you are still doing a nested loop so the requirement is still needed.

An alternative could be to have a incrementing counter instead, which might be easier to grok.

Reply:
Thanks @acodeninja. Definitely worth a refactor for grokability.